lib/probe/http-client.js

```diff
@@ -247,9 +247,14 @@ class HttpRequestMetrics {
         // ...otherwise count actual response size as its streamed
         if (isNaN(this.responseSize)) {
             this.responseSize = 0;
-            response.on("data", (chunk) => {
-                debug("⚡️ [response] data", chunk);
-                this.responseSize += chunk.length;
+
+            // resume fires when when a consumer starts consuming the stream
+            // defers response size counting to a point where a downstream consumer starts consuming
+            response.on('resume', () => {
```

Comment:
I am not sure, but this approach might still loose events if data is available when `resume` happens but before the `data` listener below is registered? In #61 I use the [response.readableFlowing](https://nodejs.org/api/stream.html#three-states) property to prevent the stream from "flowing" when we register the non-obtrusive data listener.